### ISSUE TITLE
Headers not actually supported

### ISSUE THREAD

**Issue by mxcl** [xing/XNGMarkdownParser#6]
Or they don't work. Of the form: "# Example Header" 

Bold etc. works, but headers don't, if they are intentionally unsupported then you should fix the README

**Comment by pietbrauer** [comment]
Hey @mxcl,

they should actually work. We have some test for that:

![](https://github.com/xing/XNGMarkdownParser/raw/master/Example/ExampleTests/ReferenceImages/ExampleTests/testHeaders%402x.png)

Please have a look at the [fixture](https://github.com/xing/XNGMarkdownParser/blob/master/Example/ExampleTests/headers.txt) and [the actual test](https://github.com/xing/XNGMarkdownParser/blob/master/Example/ExampleTests/SnapshotTests.m#L68-L74).

**Comment by gskbyte** [comment]
Hey @mxcl, do you have any example that showcases this (possible) bug? If it's a bug, it would be very helpful.

**Comment by mxcl** [comment]
K, I'll construct what I had.

**Comment by mxcl** [comment]
The String was `@"# Welcome\n\nFoo Bar Foo Bar" I used the singleton standard parser.

I was using the latest pod in a Swift project.

**Comment by gskbyte** [comment]
Hi, I have been checking your issue and yes, the parser does not recognize headers with backslashes. But the thing is that trying to use ``\n`` inside a header is not translated as a jump, here is what GitHub's parser does: 

![screen shot 2015-03-26 at 17 17 37](https://cloud.githubusercontent.com/assets/1763948/6850916/33aaa54a-d3dc-11e4-8f70-18e64e941d83.png)

**Comment by mxcl** [comment]
So… how am I meant to split out headers and paragraphs of text?

**Comment by pietbrauer** [comment]
@mxcl A testable scenario would help to reproduce this issue. Are you using it from JSON?

We are using it like that:

```objc
NSString *mDown = [JSON xng_dictForKey:@"markdown_text"];
if (mDown.length) {
        XNGMarkdownParser *parser = [[XNGMarkdownParser alloc] init];
        NSAttributedString *attributedString = [parser attributedStringFromMarkdownString: mDown];
        return attributedString;
}
return nil;
```

And here is how @artsy is using it in a Swift project. https://github.com/artsy/eidolon/blob/600567177ce55aed3c4b745780e8abb6287c9e10/Kiosk/App/MarkdownParser.swift
https://github.com/artsy/eidolon/blob/5bd72b7c7dc8ba0095ab231553f7dd727a9f8db9/Kiosk/Sale%20Artwork%20Details/SaleArtworkDetailsViewController.swift#L284

I hope this helps, both are retrieving it from a JSON API afaik so line breaks might occur.

**Comment by mxcl** [comment]
This is what I did:

```objc
XNGMarkdownParser *parser = [XNGMarkdownParser new];
return [parser attributedStringFromMarkdownString:@"# Welcome\n\nFoo Bar Foo Bar"];
```

**Comment by pietbrauer** [comment]
Fixed with #11
